app/scripts/lib/rpc-method-middleware/handlers/watch-asset.js

```diff
@@ -36,9 +37,11 @@ async function watchAssetHandler(
 ) {
   try {
     const { options: asset, type } = req.params;
-    res.result = await handleWatchAssetRequest(asset, type);
+    const handleWatchAssetResult = await handleWatchAssetRequest(asset, type);
+    const result = await handleWatchAssetResult.result;
+    res.result = Boolean(result);
```

Comment:
Nit: According to the spec, this should _always_ either return `true` or return an error. Casting the result as a boolean is _almost_ the same thing because the result in this case is the asset address, which should be guaranteed by our validation to be a non-empty string. But why bother obscure the expectation here.

```suggestion
    res.result = true;
```